components/script/script_thread.rs

```diff
@@ -1198,19 +1198,18 @@ impl ScriptThread {
             image_cache_thread: self.image_cache_thread.clone(),
             content_process_shutdown_chan: content_process_shutdown_chan,
             layout_threads: layout_threads,
-        };
-
-        let parent_window = self.documents.borrow().find_window(parent_pipeline_id)
-            .expect("ScriptThread: received a layout for a parent pipeline not in this script thread. This is a bug.");
+        });
 
-        // Tell layout to actually spawn the thread.
-        parent_window.layout_chan()
-                     .send(message::Msg::CreateLayoutThread(layout_creation_info))
-                     .unwrap();
+        // Pick a layout thread, any layout thread
+        match self.documents.borrow().iter().next() {
+            None => panic!("Layout attached to empty script thread."),
+            // Tell the layout thread factory to actually spawn the thread.
+            Some((_, document)) => document.window().layout_chan().send(msg).unwrap(),
```

Comment:
We could keep trying other threads if sending to the first one fails. Maybe file as a followup?